static/app/utils/profiling/gl/utils.ts

```diff
@@ -383,6 +384,27 @@ function getContext(canvas: HTMLCanvasElement, context: string): RenderingContex
 
 export {getContext};
 
+export function computeClampedConfigView(
+  newConfigView: Rect,
+  width: {max: number; min: number},
+  height: {max: number; min: number},
+  inverted: boolean
+) {
+  const clampedWidth = clamp(newConfigView.width, width.min, width.max);
+  const clampedHeight = clamp(newConfigView.height, height.min, height.max);
+
+  const maxX = width.max - clampedWidth;
+  const maxY =
+    clampedHeight >= height.max + (inverted ? 1 : 0)
+      ? 0
+      : height.max - clampedHeight + (inverted ? 1 : 0);
+
+  const clampedX = clamp(newConfigView.x, 0, maxX);
+  const clampedY = clamp(newConfigView.y, 0, maxY);
```

Comment:
It essentially does edge detection - imagine you have a configView with x:0, y:0, width: 50, height: 50 and a configSpace of x:0, y:0, width:100, height:100, then your configView X cannot go above 50, because you need to respect 0 <= configview.x + configview.width <= configspace.width (assuming configspace.x is at 0, else we would need to look at configspace.right). In the same idea, you need to respect 0 <= configview.y + configview.height <= configspace.height